src/memory.rs

```diff
@@ -585,3 +591,43 @@ impl UndefMask {
 fn bit_index(bits: usize) -> (usize, usize) {
     (bits / BLOCK_SIZE, bits % BLOCK_SIZE)
 }
+
+trait ReadBytesExt2 {
```

Comment:
Can you make these below methods free functions instead of using extension traits? I find the traits a little messy and the method syntax isn't that important.